datadog/resource_datadog_service_level_objective_test.go

```diff
@@ -228,18 +228,33 @@ func TestAccDatadogServiceLevelObjective_InvalidMonitor(t *testing.T) {
 
 // helpers
 
+func testAccCheckDatadogServiceLevelObjectiveDestroy(accProvider *schema.Provider) func(*terraform.State) error {
+	return func(s *terraform.State) error {
+		providerConf := accProvider.Meta().(*ProviderConfiguration)
+		datadogClientV1 := providerConf.DatadogClientV1
+		authV1 := providerConf.AuthV1
+		if err := destroyServiceLevelObjectiveHelper(s, authV1, datadogClientV1); err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
 func destroyServiceLevelObjectiveHelper(s *terraform.State, authV1 context.Context, datadogClientV1 *datadogV1.APIClient) error {
-	for _, r := range s.RootModule().Resources {
-		if r.Primary.ID != "" {
-			if _, _, err := datadogClientV1.ServiceLevelObjectivesApi.GetSLO(authV1, r.Primary.ID).Execute(); err != nil {
-				if strings.Contains(strings.ToLower(err.Error()), "not found") {
-					continue
+	Retry(2, 5, func() (bool, error) {
+		for _, r := range s.RootModule().Resources {
+			if r.Primary.ID != "" {
+				if _, _, err := datadogClientV1.ServiceLevelObjectivesApi.GetSLO(authV1, r.Primary.ID).Execute(); err != nil {
+					if strings.Contains(strings.ToLower(err.Error()), "not found") {
```

Comment:
If you can take the change to check the HTTP response for 404 instead that'd be cool.